### PR TITLE
feat(extension): use new trezor package of SDK in Lace [LW-6522]

### PR DESCRIPTION
**Pull request by DominikGuzei**
# Checklist

- [x] https://input-output.atlassian.net/browse/LW-6522
- [x] Wait for official npm release of new Trezor package
- [x] Update this PR to include the official new Trezor package

---

## Proposed solution

This PR is a playground to test the new Trezor hardware package we developed in the SDK in Lace.

## Testing

You need the Lace and SDK repositories locally on your machine.

1. set `USE_TREZOR_HW=true` in your local `.env` to enable Trezor hardware integration
2. Restore a hardware wallet using the Trezor flow
3. Send funds to your new hardware wallet
4. Send funds from your new hardware wallet

## Screenshots

CleanShot 2023-08-25 at 16 38 23

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2569/6545272156/index.html) for [fed9054f](https://github.com/input-output-hk/lace/pull/451/commits/fed9054fca6070a21fadc0003b840967ca85841c)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 0      | 32     | 0       | 0     | 32    | ❌     |

### CODE REVIEW

File: apps/browser-extension-wallet/.env.defaults [hunk 1]
```diff
@@ -59,17 +59,22 @@ POSTHOG_DEV_TOKEN_PREVIEW=phc_e8SaOOWpXpNE59TnpLumeUjWm4iv024AWjhQqU406jr
 CARDANO_SERVICES_URL_MAINNET=https://backend.live-mainnet.eks.lw.iog.io
 CARDANO_SERVICES_URL_PREPROD=https://backend.live-preprod.eks.lw.iog.io
 CARDANO_SERVICES_URL_PREVIEW=https://backend.live-preview.eks.lw.iog.io
+# TODO: update this with a valid sanchonet url
+CARDANO_SERVICES_URL_SANCHONET=https://backend.live-preprod.eks.lw.iog.io
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/.env.defaults:63]: Unit tests are failing - we need to add this env var to https://github.com/input-output-hk/lace/blob/main/apps/browser-extension-wallet/test/__mocks__/set-env-vars.js as well

File: apps/browser-extension-wallet/.env.defaults [hunk 2]
```diff
@@ -59,17 +59,22 @@ POSTHOG_DEV_TOKEN_PREVIEW=phc_e8SaOOWpXpNE59TnpLumeUjWm4iv024AWjhQqU406jr
...
 CEXPLORER_URL_PREPROD=https://preprod.cexplorer.io
+CEXPLORER_URL_TESTNET=https://testnet.cexplorer.io
 CEXPLORER_URL_SANCHONET=https://sanchonet.cexplorer.io
 
 # ADA Handle URLs
 ADA_HANDLE_URL_MAINNET=https://api.handle.me
 ADA_HANDLE_URL_PREVIEW=https://preview.api.handle.me
 ADA_HANDLE_URL_PREPROD=https://preprod.api.handle.me
+# TODO: update this with a valid sanchonet url
+ADA_HANDLE_URL_SANCHONET=https://preprod.api.handle.me
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/.env.defaults:77]: And possibly that one should go to `set-env-vars.js` as well

File: apps/browser-extension-wallet/.env.defaults (hunk 1 above)
Comment (DominikGuzei) [reply apps/browser-extension-wallet/.env.defaults:63]: fixed in https://github.com/input-output-hk/lace/pull/451/commits/dfdebada5be26e08f7470b1a69eeb6dcefe7d86a

File: apps/browser-extension-wallet/.env.defaults (hunk 2 above)
Comment (DominikGuzei) [reply apps/browser-extension-wallet/.env.defaults:77]: fixed in https://github.com/input-output-hk/lace/pull/451/commits/dfdebada5be26e08f7470b1a69eeb6dcefe7d86a

File: .github/workflows/e2e-tests-linux.yml [hunk 3]
```diff
@@ -50,6 +50,9 @@ jobs:
         uses: ./.github/shared/build
         with:
           LACE_EXTENSION_KEY: ${{ secrets.MANIFEST_PUBLIC_KEY }}
+          CARDANO_SERVICES_URL_MAINNET: ${{ secrets.CARDANO_SERVICES_URL_MAINNET }}
+          CARDANO_SERVICES_URL_PREPROD: ${{ secrets.CARDANO_SERVICES_URL_PREPROD }}
+          CARDANO_SERVICES_URL_PREVIEW: ${{ secrets.CARDANO_SERVICES_URL_PREVIEW }}
```
Comment (marcin-mazurek) [inline .github/workflows/e2e-tests-linux.yml:55]: @DominikGuzei I think you don't need to set those values explicitly, they should be taken from `.env.defaults`. Perhaps that's why the build is failing.

File: .github/workflows/smoke-tests.yml [hunk 4]
```diff
@@ -26,6 +26,9 @@ jobs:
         uses: ./.github/shared/build
         with:
           LACE_EXTENSION_KEY: ${{ secrets.MANIFEST_PUBLIC_KEY }}
+          CARDANO_SERVICES_URL_MAINNET: ${{ secrets.CARDANO_SERVICES_URL_MAINNET }}
+          CARDANO_SERVICES_URL_PREPROD: ${{ secrets.CARDANO_SERVICES_URL_PREPROD }}
+          CARDANO_SERVICES_URL_PREVIEW: ${{ secrets.CARDANO_SERVICES_URL_PREVIEW }}
```
Comment (marcin-mazurek) [inline .github/workflows/smoke-tests.yml:31]: @DominikGuzei again, I think you don't need to set those values explicitly, they should be taken from `.env.defaults`

File: .github/shared/build/action.yml [hunk 5]
```diff
@@ -42,6 +54,10 @@ runs:
       shell: bash
       env:
         LACE_EXTENSION_KEY: ${{ inputs.LACE_EXTENSION_KEY }}
+        CARDANO_SERVICES_URL_MAINNET: ${{ inputs.CARDANO_SERVICES_URL_MAINNET }}
```
Comment (marcin-mazurek) [inline .github/shared/build/action.yml:57]: @DominikGuzei same story - I think you don't need to set those values explicitly, they should be taken from `.env.defaults`

File: .github/workflows/e2e-tests-linux.yml (hunk 3 above)
Comment (DominikGuzei) [reply .github/workflows/e2e-tests-linux.yml:55]: ok 👍 i copied that from the sanchonet branch, so i didn't think much about it

**Comment by bslabiak** [comment]
Defect: Switching pools transaction is missing
1. Switch staking to another pool
2. View Activity tab
Results:
There is no `Delegation` trasnaction with pool detials

**Comment by DominikGuzei** [comment]
> Defect: Switching pools transaction is missing (for first initial staking it is working correctly)
> 
> 1. Switch staking to another pool
> 2. View Activity tab
>    Results:
>    There is no `Delegation` trasnaction with pool detials

@bslabiak and this is confirmed to work correctly on `main`?

**Review by tomislavhoracek — APPROVED** [review]
💯

**Review by bslabiak — APPROVED** [review]

**Review by mchappell — APPROVED** [review]
Nice work, Kudos on the tidying up along the way 🙇 🚀

**Pull request merged**